app/models/service_template.rb

```diff
@@ -286,4 +291,31 @@ def validate_order
     service_template_catalog && display
   end
   alias orderable? validate_order
+
+  private
+
+  def set_provision_action(config_info, dialog)
+    fqname = if config_info['fqname'].nil?
+               self.class.default_provisioning_entry_point(service_type)
+             else
+               config_info['fqname']
+             end
+    resource_actions.build(:action => 'Provision', :fqname => fqname, :dialog => dialog)
+    save!
+  end
+
+  def set_retire_reconfigure_actions(config_info, dialog)
+    [
+      {:name => 'Reconfigure', :param_key => 'reconfigure_fqname', :method => 'default_reconfiguration_entry_point'},
```

Comment:
You can combine two methods by adding `:args => []` to the hash, so it looks like
```
{:name => 'Provision', :param_key..., :method ..., :args => [service_type]},
{:name => 'Reconfigure', :param_key..., :method..., :args =>[]},
...
```
Then `self.class.send(action[:method], *action[:args])`